packages/app/src/pages/PoolPage/AddLiquidity.tsx

```diff
@@ -230,6 +236,28 @@ export default function AddLiquidity() {
 
   const errorsCreatePull = validateCreatePool();
 
+  const liquidityFactor = BigInt(
```

Comment:
I don't know but I guess it's a lot of calculations to have inside the component, could you maybe move this to a hook? This will make it easier for you make test them.